LayoutTests/resources/ui-helper.js

```diff
@@ -2297,6 +2297,16 @@ window.UIHelper = class UIHelper {
             testRunner.runUIScript("uiController.keyboardUpdateForChangedSelectionCount", resolve);
         });
     }
+
+    static async typeCharacters(stringToType, waitForEvent = "input", eventTarget = null) {
+        for (let character of [...stringToType]) {
+            await UIHelper.callFunctionAndWaitForEvent(async () => {
+                if (window.testRunner)
```

Comment:
Is `if (window.testRunner)` really necessary?

Reply:
Ah, I included that to match existing behavior of some tests, which are robust in the case where the test is being run manually:

```
        for (let character of [...stringToType]) {
            await UIHelper.callFunctionAndWaitForEvent(() => {
                return window.testRunner ? UIHelper.typeCharacter(character) : Promise.resolve();
            }, element, "input");
        }
```

The idea is that this helper could allow someone to still run the test manually by typing the exact string, since the test would just wait for all of the key or input events to be handled.